gobblin-core/src/main/java/gobblin/writer/http/SalesForceRestWriterBuilder.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.writer.http;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import lombok.Getter;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.JsonObject;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.converter.http.RestEntry;
+import gobblin.password.PasswordManager;
+import gobblin.writer.DataWriter;
+import gobblin.writer.http.SalesforceRestWriter.Operation;
+
+/**
+ * Builder class that builds SalesForceRestWriter where it takes connection related parameter and type of operation along with the parameters
+ * derived from AbstractHttpWriterBuilder
+ */
+public class SalesForceRestWriterBuilder extends AbstractHttpWriterBuilder<Void, RestEntry<JsonObject>, SalesForceRestWriterBuilder>{
+  private static final Logger LOG = LoggerFactory.getLogger(SalesForceRestWriterBuilder.class);
+
+  static final String SFDC_PREFIX = "salesforce.";
+  static final String CLIENT_ID = SFDC_PREFIX + "client_id";
+  static final String CLIENT_SECRET = SFDC_PREFIX + "client_secret";
+  static final String USER_ID = SFDC_PREFIX + "user_id";
+  static final String PASSWORD = SFDC_PREFIX + "password";
+  static final String SFDC_ENCRYPT_KEY_LOC = SFDC_PREFIX + ConfigurationKeys.ENCRYPT_KEY_LOC;
+  static final String USE_STRONG_ENCRYPTION = SFDC_PREFIX + "strong_encryption";
+  static final String SECURITY_TOKEN = SFDC_PREFIX + "security_token";
+  static final String OPERATION = SFDC_PREFIX + "operation";
+  static final String BATCH_SIZE = SFDC_PREFIX + "batch_size";
+  static final String BATCH_RESOURCE_PATH = SFDC_PREFIX + "batch_resource_path";
+
+  private static final Config FALLBACK;
+  static {
+    Map<String, String> configMap = ImmutableMap.<String, String>builder()
+                                                .put(AbstractHttpWriterBuilder.STATIC_SVC_ENDPOINT, "https://login.salesforce.com/services/oauth2/token")
+                                                .put(SECURITY_TOKEN, "")
+                                                .put(BATCH_SIZE, "1")
+                                                .build();
+    FALLBACK = ConfigFactory.parseMap(configMap);
+  }
+
+  @Getter private String clientId;
```

Comment:
You can have `@Getter` before the class name to avoid having a `@Getter` for each of them.